src/main/java/ch/njol/skript/expressions/ExprMaxPlayers.java

```diff
@@ -0,0 +1,137 @@
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.Bukkit;
+import org.bukkit.event.Event;
+import org.bukkit.event.server.ServerListPingEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+import com.destroystokyo.paper.event.server.PaperServerListPingEvent;
+import ch.njol.skript.ScriptLoader;
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Max Players")
+@Description({"The count of max players. This can be changed in a <a href='events.html#server_list_ping'>server list ping</a> event only.",
+		"'real max players' returns the real count of max players of the server always and can't be changed."})
+@Examples({"on server list ping:",
+		"	set the max players count to (online players count + 1)"})
+@Since("INSERT VERSION")
+public class ExprMaxPlayers extends SimpleExpression<Number> {
+
+	static {
+		Skript.registerExpression(ExprMaxPlayers.class, Number.class, ExpressionType.PROPERTY,
+				"[the] [(1¦(real|default)|2¦(fake|shown|displayed))] max[imum] player[s] [(count|amount|number|size)]",
+				"[the] [(1¦(real|default)|2¦(fake|shown|displayed))] max[imum] (count|amount|number|size) of players");
+	}
+
+	private static final boolean PAPER_EVENT_EXISTS = Skript.classExists("com.destroystokyo.paper.event.server.PaperServerListPingEvent");
+
+	private boolean isReal;
+
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		boolean isServerPingEvent = ScriptLoader.isCurrentEvent(ServerListPingEvent.class) ||
+				(PAPER_EVENT_EXISTS && ScriptLoader.isCurrentEvent(PaperServerListPingEvent.class));
+		if (parseResult.mark == 2 && !isServerPingEvent) {
+			Skript.error("The 'shown' max players count expression can't be used outside of a server list ping event");
+			return false;
+		}
+		isReal = (parseResult.mark == 0 && !isServerPingEvent) || parseResult.mark == 1;
+		return true;
+	}
+
+	@Override
+	@Nullable
+	public Number[] get(Event e) {
+		if (isReal)
+			return CollectionUtils.array(Bukkit.getMaxPlayers());
+		else
+			return CollectionUtils.array(((ServerListPingEvent) e).getMaxPlayers());
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (!isReal) {
+			if (ScriptLoader.hasDelayBefore.isTrue()) {
```

Comment:
better to use if (ScriptLoader.hasDelayBefore == Kleenean.TRUE) as it is null safe

Reply:
Not related to this PR

Reply:
neither is changing the docs :)